src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2687,7 +2687,15 @@ def update_vmss(cmd, resource_group_name, name, license_type=None, no_wait=False
                 scale_in_policy=None, priority=None, max_price=None, proximity_placement_group=None,
                 enable_automatic_repairs=None, automatic_repairs_grace_period=None, **kwargs):
     vmss = kwargs['parameters']
-    client = _compute_client_factory(cmd.cli_ctx)
+    aux_subscriptions = None
+    if vmss.virtual_machine_profile and vmss.virtual_machine_profile.storage_profile and \
+            vmss.virtual_machine_profile.storage_profile.image_reference and \
+            vmss.virtual_machine_profile.storage_profile.image_reference.id:
+        from msrestazure.tools import is_valid_resource_id, parse_resource_id
+        if is_valid_resource_id(vmss.virtual_machine_profile.storage_profile.image_reference.id):
+            res = parse_resource_id(vmss.virtual_machine_profile.storage_profile.image_reference.id)
+            aux_subscriptions = [res['subscription']]
+    client = _compute_client_factory(cmd.cli_ctx, aux_subscriptions=aux_subscriptions)
```

Comment:
what will happen if the subs id equals to ctx subs id? the current one?

Reply:
Good question. It will still add aux tenant token in the header. We can actually do this it smarter. The reason why not do this is to align implement with existing code. And there is no impact to add this if it is same with current subs, service will not reject it.

Reply:
ok